### PR TITLE
Shiny updates core patch

### PR DESCRIPTION
**Pull request by swissspidy**
Cheatin', uh?

I created this PR in order to easily create a diff file for the Shiny Updates feature plugin and to have Travis run our tests automatically.

Please ignore 🙂

More information:
-  [Core merge proposal](https://make.wordpress.org/core/2016/06/02/proposal-more-shiny-updates/)
- [Plugin on WordPress.org](https://wordpress.org/plugins/shiny-updates/)

### CODE REVIEW

File: src/wp-admin/js/updates.js [hunk 1]
```diff
@@ -305,6 +305,10 @@
 				$menuItem  = $( '#menu-appearance' );
 				$itemCount = $menuItem.find( '.theme-count' );
 				break;
+
+			default:
+				window.console.error( '"%s" is not white-listed to have its count decremented.', type );
```
Comment (swissspidy) [inline src/wp-admin/js/updates.js:310]: Attenzione signore. `wp.updates.decrementCount` is called by `wp.updates.updateItemSuccess`, which handles any type (translation updates need to be decremented too!). You should be able to pass all valid types to that function, but for 'plugin' and 'theme' it does some more stuff.

Comment (ocean90) [inline src/wp-admin/js/updates.js:310]: `window.console.error` should be wrapped by `if ( window.console && window.console.error ) { }`.

File: tests/qunit/index.html [hunk 2]
```diff
@@ -59,6 +61,7 @@
 		<script src="wp-admin/js/customize-nav-menus.js"></script>
 		<script src="wp-admin/js/customize-widgets.js"></script>
 		<script src="wp-admin/js/word-count.js"></script>
+		<script src="wp-admin/js/tests.js"></script>
```
Comment (aaronjorbin) [inline tests/qunit/index.html]: This name should match the file in wp-admin that is being tested. I believe `updates.js` is what is being tested here.

File: src/wp-admin/css/list-tables.css [hunk 3]
```diff
@@ -1678,6 +1637,14 @@ div.action-links,
 	width: 80px;
 }
 
+.wordpress-reinstall-card {
+	max-width: 100%;
+}
+
+.wordpress-reinstall-card h2 {
+	margin-top: 1em;
+}
```
Comment (ocean90) [inline src/wp-admin/css/list-tables.css]: This should be removed too.

File: src/wp-admin/includes/plugin-install.php [hunk 4]
```diff
@@ -435,6 +435,8 @@ function install_plugin_install_status($api, $loop = false) {
 function install_plugin_information() {
 	global $tab;
 
+	remove_action( 'install_plugins_pre_plugin-information', 'install_plugin_information' );
+
```
Comment (ocean90) [inline src/wp-admin/includes/plugin-install.php]: Is this still needed?

File: src/wp-admin/includes/update.php [hunk 5]
```diff
@@ -309,65 +309,69 @@ function wp_plugin_update_rows() {
...
+		echo '<tr class="plugin-update-tr' . $active_class . '" id="' . esc_attr( $response->slug . '-update' ) . '" data-slug="' . esc_attr( $response->slug ) . '" data-plugin="' . esc_attr( $file ) . '"><td colspan="' . esc_attr( $wp_list_table->get_column_count() ) . '" class="plugin-update colspanchange"><div class="update-message notice inline notice-warning notice-alt"><p>';
 
 		if ( ! current_user_can( 'update_plugins' ) ) {
-			/* translators: 1: plugin name, 2: details URL, 3: accessibility text, 4: version number */
-			printf( __( 'There is a new version of %1$s available. <a href="%2$s" class="thickbox open-plugin-details-modal" aria-label="%3$s">View version %4$s details</a>.' ),
+			/* translators: 1: plugin name, 2: details URL, 3: escaped plugin name, 4: version number */
+			printf( __( 'There is a new version of %1$s available. <a href="%2$s" class="thickbox open-plugin-details-modal" aria-label="View %3$s version %4$s details">View version %4$s details</a>.' ),
```
Comment (ocean90) [inline src/wp-admin/includes/update.php]: Was this a merge conflict or why is the accessibility text moved back to the string?

File: tests/qunit/fixtures/updates.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+};
+window._wpUpdatesItemCounts = {
+	plugins: {},
+	totals: {}
+};
+window._wpUtilSettings = {
+	'ajax': {
+		'url': '\/wp-admin\/admin-ajax.php'
+	}
+};
```
Comment (aaronjorbin) [inline tests/qunit/fixtures/updates.js:53]: You can add the global `pagenow` here.

Comment (obenland) [inline tests/qunit/fixtures/updates.js:53]: Adding it there, does it fix the errors for you? What did you set it to?

**Pull request closed without merging**